theano/sandbox/cuda/cusolver.py

```diff
@@ -0,0 +1,220 @@
+                             Work.gpudata, Lwork, devInfo.gpudata)
+            outputs[0][0] = dimshuffle(A_copy, (1, 0))
+            outputs[1][0] = TAU
+            outputs[2][0] = Work
+            outputs[3][0] = devInfo
+
+        return thunk
+
+
+gpu_geqrf = GpuGeqrf()
+
+
+class GpuOrmqr(CuSolverOp):
+    """Wrapper of cuSOLVER's implementation of the ormqr LAPACK method.
+
+    Notes
+    -----
+    Wrapping of the methods is done as in `scikits.cuda.cusolver`. The
+    interface is intendend to mimick `scipy.linalg.lapack`.
+
+    """
+    __props__ = ()
+
+    def __init__(self, side, trans, Lwork):
+        self.side = side
+        self.trans = trans
+        self.Lwork = Lwork
+        super(GpuOrmqr, self).__init__()
+
+    def output_type(self):
+        return [CudaNdarrayType(broadcastable=[False, False])(),
+                CudaNdarrayType(broadcastable=[False])(),
+                CudaNdarrayType(broadcastable=[])()]
+
+    def make_node(self, inp1, inp2, inp3):
+        inp1 = as_cuda_ndarray_variable(inp1)
+        inp2 = as_cuda_ndarray_variable(inp2)
+        inp3 = as_cuda_ndarray_variable(inp3)
+
+        assert inp1.ndim == 2
+        assert inp2.ndim == 1
+        assert inp3.ndim == 2
+
+        return theano.Apply(self, [inp1, inp2, inp3], self.output_type())
+
+    def _make_thunk(self, node, storage_map, _, no_recycling=[]):
+        inputs = [storage_map[v] for v in node.inputs]
+        outputs = [storage_map[v] for v in node.outputs]
+
+        def thunk():
+            A = inputs[0][0]
+            TAU = inputs[1][0]
+            C = inputs[2][0]
+            lda, k = A.shape
+            m, n = C.shape
+            ldc = max(1, m)
+
+            # LAPACK routines expect Fortran order
+            A_T = dimshuffle(A, (1, 0))
+            A_copy = A_T.copy()
+            C_T = dimshuffle(C, (1, 0))
+            C_copy = C_T.copy()
+
+            assert self.side in ('L', 'R')
+            assert self.trans in ('N', 'T')
+            assert ldc >= max(1, m)
+            if self.side == 'L':
+                assert lda >= max(1, m)
+                assert self.Lwork >= max(1, n)
+                assert m >= k >= 0
+            else:
+                assert lda >= max(1, n)
+                assert self.Lwork >= max(1, m)
+                assert n >= k >= 0
+
+            Work = CudaNdarray.zeros((self.Lwork,))
+            devInfo = CudaNdarray.zeros((1,))
+
+            cusolverDnSormqr(handle, _CUBLAS_SIDE_MODE[self.side],
+                             _CUBLAS_OP[self.trans], m, n, k,
+                             A_copy.gpudata, lda, TAU.gpudata, C_copy.gpudata,
+                             ldc, Work.gpudata, self.Lwork, devInfo.gpudata)
+
+            outputs[0][0] = dimshuffle(C_copy, (1, 0))
+            outputs[1][0] = Work
+            outputs[2][0] = devInfo
+
+        return thunk
+
+
+def gpu_ormqr(side, trans, a, tau, c, lwork):
```

Comment:
A short documentation for the arguments that, this function requires might be useful.